infra/cifuzz/fuzz_target.py

```diff
@@ -232,17 +251,67 @@ def download_oss_fuzz_build(self):
     if not latest_build_str:
       return None
 
-    oss_fuzz_build_url = url_join(GCS_BASE_URL, self.project_name,
-                                  latest_build_str)
+    oss_fuzz_build_url = url_join(GCS_BASE_URL, CLUSTERFUZZ_BUILDS,
+                                  self.project_name, latest_build_str)
+    return download_and_unpack_zip(oss_fuzz_build_url, build_dir)
+
+  def download_latest_corpus(self):
+    """Downloads the latest OSS-Fuzz corpus for the target from google cloud.
+
+    Returns:
+      The local path to to corpus or None if download failed.
+    """
+    if not self.project_name:
+      return None
+    if not os.path.exists(self.out_dir):
+      logging.error('Out directory %s does not exist.', self.out_dir)
+      return None
+
+    corpus_dir = os.path.join(self.out_dir, 'backup_corpus', self.target_name)
+    os.makedirs(corpus_dir, exist_ok=True)
+    project_qualified_fuzz_target_name = self.target_name
+    qualified_name_prefix = '%s_' % self.project_name
+    if not self.target_name.startswith(qualified_name_prefix):
+      project_qualified_fuzz_target_name = qualified_name_prefix + \
+      self.target_name
+    corpus_url = url_join(
```

Comment:
nit: this URL logic below is pretty hard to read, I'd define another global variable e.g.

```python
CORPUS_BACK_URL = (
    GCS_BASE_URL + '/{project}-backup.clusterfuzz-external.appspot.com/'
    'corpus/libFuzzer/{fuzz_target}/public.zip')
```

and then used it:

```python
  CORPUS_BACK_URL.format(project=self.project_name,
                         fuzz_target=project_qualified_fuzz_target_name)
```

I don't think that it makes a lot of sense to rely on `url_join`, as we have a bunch of `/` characters typed in the constant part, also `CORPUS_ZIP_NAME` is used only ones, so it's fine to remove it and just have as a part of the URL